syntax = "proto3";

package orchestrator;

message Task {
    enum TaskType {
        TASK_TYPE_FULL = 0;
        TASK_TYPE_DELTA = 1;
    }

    string message_id = 1;
    int64 message_time = 2;
    string task_id = 3;
    TaskType task_type = 4;
    repeated SourceDetail sources = 5;
    int64 deadline = 6;
    bool cancelled = 7;
}

message SourceDetail {
    enum SourceType {
        SOURCE_TYPE_REPO = 0;
        SOURCE_TYPE_FUZZ_TOOLING = 1;
        SOURCE_TYPE_DIFF = 2;
    }

    string sha256 = 1;
    SourceType source_type = 2;
    string url = 3;
    optional string path = 4;
}

message TaskDownload {
    Task task = 1;
}

message TaskReady {
    Task task = 1;
}

message TaskDelete {
    string task_id = 1;
    float received_at = 2;
}

message TaskVulnerability {
    string task_id = 1;
    string vulnerability_id = 2;
    string package_name = 3;
    string sanitizer = 4;
    string harness_path = 5;
    string data_file = 6;
    string architecture = 7;
}

message Patch {
    string task_id = 1;
    string vulnerability_id = 2;
    string patch = 3;
}
